capstone/src/test/java/com/google/sps/servlets/AllGroupMembersServletTest.java

```diff
@@ -0,0 +1,117 @@
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.Objects.User;
+import com.google.sps.Objects.Badge;
+
+public class AllGroupMembersServletTest {
+
+  private static final String USER_EMAIL = "test@test.com";
+  private static final String USER_ID = "test";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig()
+                  .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+              new LocalUserServiceTestConfig())
+          .setEnvEmail(USER_EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail.com")
+          .setEnvAttributes(
+              new HashMap(
+                  ImmutableMap.of(
+                      "com.google.appengine.api.users.UserService.user_id_key", USER_ID)));
+
+  private final User USER_1 =  new User("22222", "Test", "McTest", 
+                          "testy@gmail.com", 
+                          /* phoneNumber= */ "123-456-7890", 
+                          /* profilePic= */ "", 
+                          /* badges= */ new LinkedHashSet<Badge>(), 
+                          /* groups= */ new LinkedHashSet<Long>(), 
+                          /* interests= */ new ArrayList<String>()
+  );
+
+  private final User USER_2 =  new User("333333", "Test", "McTest2", 
+                          "testy2@gmail.com", 
+                          /* phoneNumber= */ "111-111-1111", 
+                          /* profilePic= */ "", 
+                          /* badges= */ new LinkedHashSet<Badge>(), 
+                          /* groups= */ new LinkedHashSet<Long>(), 
+                          /* interests= */ new ArrayList<String>()
+  );
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private DatastoreService datastore;
+  private AllGroupMembersServlet allGroupMembersServlet;
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Add test data
+    ImmutableList.Builder<Entity> users = ImmutableList.builder();
+    users.add(USER_1.toEntity());
+    users.add(USER_2.toEntity());
+    datastore.put(users.build());
+
+    // Set up a fake HTTP response.
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+    allGroupMembersServlet = new AllGroupMembersServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    responseWriter = null;
+    datastore = null;
+    allGroupMembersServlet = null;
```

Comment:
These are unnecessary because setUp is called before every test so it will make sure to return everything to the state that you configured them in setUp